aggregator/src/aggregator.rs

```diff
@@ -3692,6 +3720,33 @@ where
 /// See: https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Access-Control-Max-Age
 const CORS_PREFLIGHT_CACHE_AGE: u32 = 24 * 60 * 60;
 
+/// Constructs a Warp filter that matches the request path for tasks, as well as whatever regex is
+/// provided, but without consuming the path or any values encoded into it, such that later filters
+/// can still extract path components.
+fn path_matching_filter(trailing_fragment_regex: &'static str) -> BoxedFilter<()> {
+    warp::path::peek()
+        .and_then(move |peek: warp::path::Peek| async move {
+            let regex = Regex::new(&format!(
```

Comment:
I'd recommend that we compile this regex at filter construction time, and return `Result<BoxedFilter<()>, regex::Error>` out instead of rejecting requests, so that we can catch mistakes in regexes early. Then, the compiled regex can be moved into the closure. (Also, I think we can swap `.and_then()` for `.map()`, since the async block here doesn't await anything)

Reply:
Good point about the regex. I used `and_then` because it's obvious how to return a rejection from it. It may be possible to do this with `Filter::map` but I am disinclined to spend further time learning `warp`'s idiosyncrasies since we plan to migrate off of it anyway.

Reply:
It might work if the closure returns `Result<(), Rejection>`, but I did not test this.